src/ContentFileMigrationTask.php

```diff
@@ -0,0 +1,277 @@
+     * Get the array of Class, Table and Field mapping for updating
+     *
+     * @return array
+     */
+    protected function getMapping()
+    {
+        if (empty($this->mapping)) {
+            $this->setMapping();
+        }
+
+        return $this->mapping;
+    }
+
+    /**
+     * Yield database fields for a given class
+     *
+     * @param $class
+     * @return \Generator
+     */
+    protected function getDBFields($class)
+    {
+        foreach ($class::singleton()->getSchema()->fieldSpecs($class) as $field => $type) {
+            yield $field => $type;
+        }
+    }
+
+    /**
+     * Get the appropriate table to update via SQL as to not publish draft content from the SS3 site
+     *
+     * @param $class
+     * @param $field
+     * @return mixed
+     */
+    protected function getFieldTable($class, $field)
+    {
+        return $class::singleton()->getSchema()->tableForField($class, $field);
+    }
+
+    /**
+     * Update records of based on ContentFileMigrationTask::mapping
+     *
+     * @param $class
+     * @param $table
+     * @param $record
+     * @param $field
+     */
+    protected function updateFileReference($class, $table, $record, $field)
+    {
+        if (isset($record[$field])) {
+            if (preg_match('/<img\s*(?:src\s*\=\s*[\'\"](.*?)[\'\"].*?\s*|\s*|\s*)+.*?>/sm', $record[$field], $matches)) {
+                foreach ($matches as $match) {
+                    preg_match('/src=".*"/', $match, $source);
+                    if (!empty($source)) {
+                        $path = substr($source[0], 5, strlen($source[0]) - 6);
+                        $parts = explode('/', $path);
+                        $file = File::get()->filter('FileFilename:PartialMatch', $parts[count($parts) - 1])->first();
+
+                        $newValue = $this->getNewFieldValue($record, $field, $match, $file);
+                        if ($newValue) {
+                            try {
+                                DB::prepared_query("UPDATE \"{$table}\" SET \"{$field}\" = ? WHERE ID = ?", [$newValue, $record['ID']]);
+                                echo "{$table}: Updated image reference for: {$class} - {$record['ID']}\n";
+                            } catch (\Exception $e) {
+                                echo "{$table}: Failed updating {$class} - {$record['ID']}\n";
+                            }
+                        } else {
+                            echo "{$table}: Couldn't update {$class} = {$record['ID']} - {$match}\n";
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Replace image instances with new shortcode
+     *
+     * @param $record
+     * @param $field
+     * @param $match
+     * @param $file
+     * @return mixed
+     */
+    protected function getNewFieldValue($record, $field, $match, $file)
+    {
+        if ($file instanceof File) {
+            $newPath = $file->getURL();
+
+            $find = [
```

Comment:
Related to some notes in the comments for the PR, I've come across the `""""` issue with empty `alt` tags as well. After some testing I think removing empty `title=""` references and generating a value for empty `alt` tags might be a good option here.